kernel/kernel-api/src/main/java/io/delta/kernel/internal/rowtracking/MaterializedRowTrackingColumn.java

```diff
@@ -193,14 +198,126 @@ public static List<StructField> convertToPhysicalColumn(
             logicalField.getName()));
   }
 
-  private String getPhysicalColumnName(Metadata metadata) {
-    return Optional.ofNullable(metadata.getConfiguration().get(getMaterializedColumnNameProperty()))
+  /**
+   * Computes row IDs and row commit versions based on their materialized values if present in the
+   * data returned by the Parquet reader, using the base row ID and default row commit version from
+   * the AddFile otherwise.
+   *
+   * @param dataBatch a batch of physical data read from the table.
+   * @param scanFile the {@link Row} representing the scan file metadata.
+   * @param configuration the configuration map containing table metadata.
+   * @param engine the {@link Engine} to use for expression evaluation.
+   * @return a new {@link ColumnarBatch} with logical row tracking columns
+   */
+  public static ColumnarBatch transformPhysicalData(
```

Comment:
cc @raveeram @vkorukanti this would be so much better with the expression fixup changes